code/game/machinery/camera/camera_assembly.dm

```diff
@@ -129,8 +129,13 @@
 			user << "The assembly already has \a [W] inside!"
 			return
 		user << "You attach the [W] into the assembly inner circuits."
-		upgrades += W
-		user.drop_item(W, src)
+		if (istype(W, /obj/item/stack/sheet/mineral/plasma))
+			var/obj/item/stack/sheet/mineral/plasma/s = W
+			s.use(1)
+			upgrades += new /obj/item/stack/sheet/mineral/plasma
+		else
+			upgrades += W
+			user.drop_item(W, src)

```

Comment:
If you change it to stack instead of the full path it only actually works once and then if you were to remove the installed plasma and apply any whatsoever, it will once more consume the entire stack, indicating it's failing the istype check, very weird.